packages/gladia/src/gladia-transcription-model.ts

```diff
@@ -0,0 +1,929 @@
+        path: '/v2/upload',
+        modelId: '',
+      }),
+      headers: combineHeaders(this.config.headers(), options.headers),
+      formData,
+      failedResponseHandler: gladiaFailedResponseHandler,
+      successfulResponseHandler: createJsonResponseHandler(
+        gladiaUploadResponseSchema,
+      ),
+      abortSignal: options.abortSignal,
+      fetch: this.config.fetch,
+    });
+
+    const { body, warnings } = await this.getArgs(options);
+
+    const { value: transcriptionInitResponse } = await postJsonToApi({
+      url: this.config.url({
+        path: '/v2/pre-recorded',
+        modelId: '',
+      }),
+      headers: combineHeaders(this.config.headers(), options.headers),
+      body: {
+        ...body,
+        audio_url: uploadResponse.audio_url,
+      },
+      failedResponseHandler: gladiaFailedResponseHandler,
+      successfulResponseHandler: createJsonResponseHandler(
+        gladiaTranscriptionInitializeResponseSchema,
+      ),
+      abortSignal: options.abortSignal,
+      fetch: this.config.fetch,
+    });
+
+    // Poll the result URL until the transcription is done or an error occurs
+    const resultUrl = transcriptionInitResponse.result_url;
+    let transcriptionResult;
+    let transcriptionResultHeaders;
+
+    while (true) {
+      const response = await getFromApi({
+        url: resultUrl,
+        headers: combineHeaders(this.config.headers(), options.headers),
+        failedResponseHandler: gladiaFailedResponseHandler,
+        successfulResponseHandler: createJsonResponseHandler(
+          gladiaTranscriptionResultResponseSchema,
+        ),
+        abortSignal: options.abortSignal,
+        fetch: this.config.fetch,
+      });
+
+      transcriptionResult = response.value;
+      transcriptionResultHeaders = response.responseHeaders;
+
+      if (
+        transcriptionResult.status === 'done' ||
+        transcriptionResult.status === 'error'
+      ) {
+        break;
+      }
+
+      // Wait for 1 second before polling again
+      await new Promise(resolve => setTimeout(resolve, 1000));
+    }
+
+    // Handle error status
+    if (transcriptionResult.status === 'error') {
+      throw new Error(
+        `Transcription failed with error code: ${transcriptionResult.error_code}`,
+      );
+    }
+
+    if (!transcriptionResult.result) {
+      throw new Error('Transcription result is empty');
+    }
+
+    // Process the successful result
+    return {
+      text: transcriptionResult.result.transcription.full_transcript,
+      durationInSeconds: transcriptionResult.result.metadata.audio_duration,
+      language: transcriptionResult.result.transcription.languages.at(0),
+      segments: transcriptionResult.result.transcription.utterances.map(
+        utterance => ({
+          text: utterance.text,
+          startSecond: utterance.start,
+          endSecond: utterance.end,
+        }),
+      ),
+      response: {
+        timestamp: currentDate,
+        modelId: '',
```

Comment:
leave undefined?

Reply:
Needs to be a string, so I'll use `default`